### ISSUE TITLE
question about the BERT-style pretraining

### ISSUE THREAD

**Issue by linjing7** [teddykoker/image-gpt#6]
Hi, teddykoker, thanks a lot for your excellent work. It help me a lot to understand the paper.
Here I have a question about the BERT-style pretraining. As I know , BERT-style pretraing is done as follow: mask some words or pixels, and then predict it. But I can't find the masking operator in the code, could you please tell me where it is?
Thanks again for your work.

**Comment by linjing-ai** [comment]
sorry, I have fould it.

**Comment by cyber-art** [comment]
I think the author coded only the auto-regressive pre-training task. I couldn't see the BERT task either. If you read the original paper of Image-GPT written by OpenAI, there are two different pre-training methods: BERT and auto-regressive. In this implementation only auto-regressive is used.

Link of original paper
https://cdn.openai.com/papers/Generative_Pretraining_from_Pixels_V2.pdf
